core/src/components/datetime/datetime.tsx

```diff
@@ -1262,9 +1262,63 @@ export class Datetime implements ComponentInterface {
        */
       this.activeParts = [];
     }
+
+    /**
+     * Only animate if:
+     * 1. We're using grid style (wheel style pickers should just jump to new value)
+     * 2. The month and/or year actually changed (otherwise there's nothing to animate to)
+     * 3. The datetime is visible (prevents animation when in collapsed datetime-button, for example)
+     * 4. The month/year picker is not open (since you wouldn't see the animation anyway)
+     */
+    const didChangeMonth = month !== workingParts.month || year !== workingParts.year;
+    const elIsVisible = el.offsetParent !== null;
+    const { isGridStyle, showMonthAndYear } = this;
+    if (animate && isGridStyle && didChangeMonth && elIsVisible && !showMonthAndYear) {
+      /**
+       * Tell other render functions that we need to force the
+       * target month to appear in place of the actual next/prev month.
+       * Because this is a State variable, a rerender will be triggered
+       * automatically, updating the rendered months.
+       */
+      this.forceRenderDate = { month, year, day };
```

Comment:
Any reason why we used the individual pieces here when we use `targetValue` below? Wondering if we should just use `targetValue` here for consistency.

Reply:
Good catch; I think this was leftover from an earlier draft where I was only passing in the month and year.